tomee/tomee-microprofile/mp-common/src/main/java/org/apache/tomee/microprofile/TomEEMicroProfileListener.java

```diff
@@ -122,7 +122,13 @@ public void processApplication(@Observes final BeforeEvent<AfterApplicationCreat
         // index only once and pass it everywhere it's needed. Also in order to build the index, we need the entire
         // application so doing it here from the AppInfo is way simpler
         try {
-            final Index index = of(afterApplicationCreated.getEvent().getApp().libs);
+            final Index index;
+            if(indexCache.containsKey(afterApplicationCreated.getEvent().getApp())) {
```

Comment:
you can do a get and test null which is always better than a double hash (contains + get)

Reply:
I changed it to computeIfAbsent